sorts/bubble_sort.py

```diff
@@ -0,0 +1,14 @@
+def bubble_sort(unorder_list: list) -> list:
+    """
+    Bubble sort, sometimes referred to as sinking sort, is a simple sorting algorithm that repeatedly steps through the list, compares adjacent elements and swaps them if they are in the wrong order. The pass through the list is repeated until the list is sorted. The algorithm, which is a comparison sort, is named for the way smaller or larger elements "bubble" to the top of the list.
+    """
+    count = len(unorder_list) # sets the lenght of the array as a count so next iteractions is sorter as it is reduced by one each cycle.
+    while count >= 2: # initiates outter iteration based on array lenght 'count'.
+        for i in range(count): # inner iteration goes through each array set by outter loop.
+            temp_i = None # a temporary variable to hold an item of the pair is a swap is required.
+            if i < count-1 and unorder_list[i] > unorder_list[i+1]: # check if a swap is requited and execute the swap.
```

Comment:
if  the array is mostly sorted  or sorted , an you avoid unnecessary  iterations.

Reply:
Please review new code additions. unnecessary iterations were eliminated.